### ISSUE TITLE
Verify Maximum Limits

### ISSUE THREAD

**Issue by alejmedinajr** [alejmedinajr/AI-Detector#72]
The following maximum limits need to be verified:

- Maximum file upload length: multiple files can be uploaded. The maximum amount of files that can be uploaded on one time need s to be defined. We may also want to decide to put a constraint on this as well.
- Maximum prompt limit: We need to find out the maximum number of characters the prompt/submission form can handle for **both** AI API's. This is where chunking can be used, or we can just decide to put a constraint on this as well. This needs to be done for uploading files (the character count) and for plain text (essentially both of our form settings).
- Maximum user email and password lengths: Google Auth may take care of this, but we need to find out the maximum number of characters for this as well, just to make sure we know what happens when it is reached.

**Comment by alejmedinajr** [comment]
Looking into this now

**Comment by alejmedinajr** [comment]
I have found the maximum limit and chatgpt api has. I also found out that there are two ways I can go about this:

1. I can install an npm package that has encoding and decoding abilities for word to token
2. I can install a similar package that does the same for python.

I think this really depends on where it would be wiser to implement this. I am going to go on a hunch and assume it would be better to implement this on the react side, considering it would be less expensive to validate the form in this manner before sending an API request, only to send it back with an error to redo the form.

**Comment by alejmedinajr** [comment]
I ended up deciding the easiest way to deal with this was to add an if statement to both api calls to the endpoints to verify that the maximum token count for a prompt is not met. This is defined as a constant variable, which is defined by the api. I ended up taking the minimum of the maximum token count for both models, which ended up being (slightly above) 100000. I used the chakra toast package to display an error message that does not allow the the endpoint to be called. Moreover, this does not crash the app, it instead displays the custom error message for a set duration of time before the user can try again with a different prompt. 
<img width="748" alt="Screenshot 2024-04-19 at 7 07 59 PM" src="https://github.com/alejmedinajr/AI-Detector/assets/100097809/4f5988d7-abee-4a60-b8ee-c45ae68b8c89">

I also tested to make sure this does not break future calls to api (it does not).
